beacon/blockchain/payload.go

```diff
@@ -263,17 +250,9 @@ func (s *Service) optimisticPayloadBuild(
 	}
 
 	// We then trigger a request for the next payload.
-	payload := blk.GetBody().GetExecutionPayload()
```

Comment:
Right before this we run `stateprocessor.Transition` which actually store the payload header to state (see `processExecutionPayload`). So it's fine dropping payload here and load latest execution payload header from state in `RequestPayloadAsync`